automation/terraform/modules/testnet-alerts/templates/testnet-alert-rules.yml.tpl

```diff
@@ -272,6 +272,15 @@ groups:
       description: "The rate of {{ $value }} new blocks observed by archive postgres instances is low on network {{ $labels.testnet }}."
       runbook: "https://www.notion.so/minaprotocol/Archive-Node-Metrics-9edf9c51dd344f1fbf6722082a2e2465"
 
+  - alert: OneOfNodeRestarted
+    expr: min by (testnet) (Coda_Runtime_process_uptime_ms_total ${rule_filter}) < 60000
```

Comment:
I don't think we get any metrics while we're starting up; if we're generating a genesis proof, this could be a minute or more. A longer period (3 mins? 5 mins?) may catch these cases better.

Reply:
It would also be nice to capture the number of peers that have restarted within the last n minutes; perhaps a query like
```promql
count by (testnet) (Coda_Runtime_process_uptime_ms_total{testnet=~"mainnet|devnet2|snappnet"} < 360000) > 0
```
[link](https://o1testnet.grafana.net/explore?orgId=1&left=%5B%22now-2d%22,%22now%22,%22grafanacloud-o1testnet-prom%22,%7B%22expr%22:%22count%20by%20(testnet)%20(Coda_Runtime_process_uptime_ms_total%7Btestnet%3D~%5C%22mainnet%7Cdevnet2%7Csnappnet%5C%22%7D%20%3C%20360000)%22,%22format%22:%22time_series%22,%22instant%22:false,%22interval%22:%22%22,%22datasource%22:%22grafanacloud-o1testnet-prom%22,%22exemplar%22:false,%22requestId%22:%22Q-4ce48336-2048-47c1-b7bc-5c38279e356b-0A%22%7D%5D) would give us better insight? It may also make sense to have a critical alert if some larger number have restarted in the time period, since e.g. 4 nodes going down within 5 minutes leaves the network in a worrying state.